src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/CourseActivityBuilder.kt

```diff
@@ -54,19 +58,57 @@ class CourseActivityBuilder(
   var payRates: List<CourseActivityPayRateBuilder>,
 ) {
   fun build(): CourseActivity =
+    repository?.let {
+      CourseActivity(
+        code = code,
+        program = repository.lookupProgramService(programId),
+        caseloadId = prisonId,
+        prison = repository.lookupAgency(prisonId),
+        description = description,
+        capacity = capacity,
+        active = active,
+        scheduleStartDate = LocalDate.parse(startDate),
+        scheduleEndDate = endDate?.let { LocalDate.parse(it) },
+        iepLevel = repository.lookupIepLevel(minimumIncentiveLevelCode),
+        internalLocation = internalLocationId?.let { repository.lookupAgencyInternalLocation(it) },
+      ).apply {
+        payRates.addAll(this@CourseActivityBuilder.payRates.map { it.build(this) })
+      }
+    }
+      ?: throw IllegalStateException("No repository - is this a unit test? Try create() instead.")
+
+  fun create(
+    programCode: String = "CA",
+    internalLocationCode: String? = "CRM1"
+  ): CourseActivity =
     CourseActivity(
       code = code,
-      program = repository.lookupProgramService(programId),
+      program = programService(programCode),
```

Comment:
In unit tests we call the `create` function instead which instantiates related entities manually (as there is no database).